src/jit/flowgraph.cpp

```diff
@@ -8064,6 +8068,16 @@ void Compiler::fgAddInternal()
 {
     noway_assert(!compIsForInlining());
 
+#ifndef LEGACY_BACKEND
+    // The RyuJIT backend requires a scratch BB into which it can safely insert a P/Invoke method prolog if one is
+    // required. Create it here.
+    if (info.compCallUnmanaged != 0)
+    {
+        fgEnsureFirstBBisScratch();
+        fgFirstBB->bbFlags |= BBF_DONT_REMOVE;
```

Comment:
OK and I guess we removed  the empty block, unless the BBF_DONT_REMOVE is set.

Maybe we should look into having fgEnsureFirstBBisScratch always set that flag, or maybe that is doing to much as we still want to merge the blocks for the other cases.